### PR TITLE
Fix for internal issue #633

### PR DESCRIPTION
**Pull request by Dronplane**
Made functions ANALYZER, BOOST and MIN_MATCH callable during optimization
Fixed determinig ArangoSearch filter volatility in cases of  non-deterministic function call in condition
Added tests.

https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/5995/

### CODE REVIEW

**Comment by Dronplane** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/6000/

File: arangod/IResearch/IResearchFeature.cpp [hunk 1]
```diff
@@ -147,6 +147,45 @@ arangodb::aql::AqlValue dummyFilterFunc(arangodb::aql::ExpressionContext*,
...
+  } else {
+    auto message = std::string("'MIN_MATCH' AQL function: ")
+                     .append(" last argument has invalid type '") 
+                     .append(minMatchValue.getTypeString())
+                     .append("' (numeric expected)");
+    THROW_ARANGO_EXCEPTION_MESSAGE(
+      TRI_ERROR_BAD_PARAMETER,
+      message);
+  }
+  return arangodb::aql::AqlValue(arangodb::aql::AqlValueHintBool(false));
```
Comment (jsteemann) [inline arangod/IResearch/IResearchFeature.cpp]: this line is unreachable.
```suggestion
```

**Comment by Dronplane** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/6003/

**Comment by Dronplane** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/6008/

**Pull request merged**